sdcm/sct_events.py

```diff
@@ -111,14 +111,18 @@ def get_client_socket(self, filter_type=b''):
         socket.setsockopt(zmq.SUBSCRIBE, filter_type)  # pylint: disable=no-member
         return socket
 
-    def subscribe_events(self, filter_type=b'', stop_event=None):
+    def subscribe_events(self, filter_type=b'', stop_event: Union[multiprocessing.Event, threading.Event] = None):
         # pylint: disable=too-many-nested-blocks,too-many-branches
         LOGGER.info("subscribe to server with port %d", self.sub_port.value)
         socket = self.get_client_socket(filter_type)
         filters = dict()
+        reiterate = False
         try:
-            while stop_event is None or not stop_event.isSet():
-                if socket.poll(timeout=1):
+            while reiterate or stop_event is None or not stop_event.is_set():
+                # If there is an object scheduled, reiterate cycle, not looking if stop_event is set
+                #   Done in order to make sure that all events end up at the files when stop() is called
+                reiterate = socket.poll(timeout=1)
+                if reiterate:
```

Comment:
What do you think about replacing this with:

```python
while True:
    if not socket.poll(timeout=1):
        if stop_event and stop_event.is_set():  # Break the cycle only if there is nothing to get from the socket.
            break
        continue
    ...
```

From my pov it's much cleaner, you don't need additional variable to control the cycle and it saves one indentation level inside the cycle.